diff --git a/build/defaultResolver.js b/build/defaultResolver.js
index a19986bc668969d793b194073e93ded46bc749e5..2927a26c6f0b460ab7604841dc91777639f668fc 100644
--- a/build/defaultResolver.js
+++ b/build/defaultResolver.js
@@ -56,6 +56,11 @@ const defaultResolver = (path, options) => {
     return (0, _jestPnpResolver().default)(path, options);
   }
 
+  // https://github.com/facebook/jest/issues/11604#issuecomment-866629115
+  if (path === 'pnpapi') {
+    return require.resolve('pnpapi', { paths: [options.basedir] });
+  }
+
   const resolveOptions = {
     ...options,
     isDirectory: _fileWalkers.isDirectory,
